src/fmt/termcolor/shim_impl.rs

```diff
@@ -0,0 +1,53 @@
+use std::io::{self, Write};
+
+use fmt::{WriteStyle, Target};
+
+pub(in ::fmt) mod pub_use_in_super {
+    
+}
+
+pub(in ::fmt) struct BufferWriter {
+    target: Target,
+}
+
+pub(in ::fmt) struct Buffer(Vec<u8>);
+
+impl BufferWriter {
+    pub(in ::fmt) fn stderr(_: WriteStyle) -> Self {
+        BufferWriter {
+            target: Target::Stderr,
+        }
+    }
+
+    pub(in ::fmt) fn stdout(_: WriteStyle) -> Self {
+        BufferWriter {
+            target: Target::Stdout,
+        }
+    }
+
+    pub(in ::fmt) fn buffer(&self) -> Buffer {
+        Buffer(Vec::new())
+    }
+
+    pub(in ::fmt) fn print(&self, buf: &Buffer) -> io::Result<()> {
+        match self.target {
```

Comment:
These should probably acquire a lock on the stream first, since it might make multiple calls to `write`.